upup/pkg/fi/cloudup/template_functions.go

```diff
@@ -156,29 +156,28 @@ func (tf *TemplateFunctions) DnsControllerArgv() ([]string, error) {
 	}
 
 	if dns.IsGossipHostname(tf.cluster.Spec.MasterInternalName) {
-		argv = append(argv, "--dns=gossip")
```

Comment:
I am confused why this is being removed.  If the hostname is gossip only then we should be using gossip.  Need some more context on this.

Reply:
According to #4055, it's not able to use public domain name for services when using gossip.

Reply:
If I'm reading it correctly though, this breaks gossip on AWS and forces you to use route53 (?)

Perhaps we need multiple DNS providers in protokube, or we could also suggest using external-dns with gossip.

Reply:
To quote @tommyvn in issue #4055 "My understanding is that gossip handles the master discovery, kube-dns handles internal DNS, and dns-controller interfaces with my chosen external dns provider to allow me to create ELBs with custom host names by creating Service resources with dns.alpha.kubernetes.io/external annotations, as per https://github.com/kubernetes/kops/blob/master/dns-controller/README.md. I realise route53 isn't appropriate for everything, like AWS CN region for eg, but I can't see this change enabling functionality for China AWS, rather it seems to have broken all the other regions interface with route53."

Are you saying that we should be able to choose between gossip or route53 for external DNS?